libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -293,9 +334,55 @@ static void infer_completion_callback(void *args)
     }
 }
 
-static int init_model_ov(OVModel *ov_model, const char *input_name, const char *output_name)
+static int load_network_vaapi(OVModel *ov_model, const AVFrame* input_frame, const char *input_name)
+{
+    // for model loaded in vaapi, we should do all prepostprocess with ov CAPI.
+    IEStatusCode status;
+    AVBufferRef* hw_device_ctx = ov_model->model->filter_ctx->hw_device_ctx;
+    AVHWDeviceContext *hw_device_data = (AVHWDeviceContext *)hw_device_ctx ->data;
+    AVVAAPIDeviceContext *va_device_data = (AVVAAPIDeviceContext *)hw_device_data ->hwctx;
+    VADisplay display = va_device_data -> display;
+    input_shapes_t input_shapes;
+    ie_config_t config = {NULL, NULL, NULL};
+    OVContext *ctx = &ov_model->ctx;
+    status = ie_network_get_input_shapes(ov_model->network, &input_shapes);
+    if (status != OK) {
+        return DNN_GENERIC_ERROR;
+    }
+    for (int i = 0; i < input_shapes.shape_num; i++) {
+        // reshape input shapes to enable auto resize algorithm
+        input_shapes.shapes[i].shape.dims[0] = ctx->options.batch_size;
+        input_shapes.shapes[i].shape.dims[2] = input_frame->height;
+        input_shapes.shapes[i].shape.dims[3] = input_frame->width;
+    }
+    status = ie_network_reshape(ov_model->network, input_shapes);
+    ie_network_input_shapes_free(&input_shapes);
+    if (status != OK) {
+        return DNN_GENERIC_ERROR;
+    }
+    // Auto resize and convert color format in openvino preprocess
+    status |= ie_network_set_input_resize_algorithm(ov_model->network, input_name, RESIZE_BILINEAR);
+    status |= ie_network_set_input_layout(ov_model->network, input_name, NCHW);
+    status |= ie_network_set_input_precision(ov_model->network, input_name, U8);
+    // set input color format to NV12 to enable automatic input color format convert to bgr,
+    // if not use ie_network_set_color_format the input blob will keep its own color format.
+    status |= ie_network_set_color_format(ov_model->network, input_name, NV12);
```

Comment:
This `status` is not checked and you assign a new return value to it.

Reply:
Fixed.